### DISCUSSION TITLE
The Copilot4Eclipse Chat can't access the ABAP Development Tools ABAP Editor Context

### DISCUSSION THREAD

**Discussion by ansgar-maehrlein** [Genuitec/Copilot4Eclipse-community#9] [Help]
1. Please describe the issue you are experiencing:
The Copilot4Eclipse Chat can't access the ABAP Development Tools ABAP Editor Context
While autocompletion suggestions work, commands like /explain do not work, even from the right-click-context-menu.

1a Describe the steps to reproduce:
-Mark code
-Right click on marked code in the ABAP Editor > Copilot4Eclipse > Explain This
-Chat Panel result: "I'm sorry, but I can't provide the explanation you're asking for because the selected code you're referring to hasn't been provided. Could you please provide the code snippet you want explained?"

I can get it to work by fiddling with Using the right-click-context-menu action "Reference File in Chat Panel" and then entering /explain in the chat panel, but this is limited to the entire file.
-Right click on the ABAP Editor > "Reference File in Chat Panel"
-enter "/explain" and send in chat panel
-Chat Panel result: "I'm sorry, but I can't provide the explanation you're asking for because the selected code you're referring to hasn't been provided. Could you please provide the code snippet you want explained?"
-Mark code
-Right click on marked code in the ABAP Editor > Copilot4Eclipse > Explain This
-Chat Panel result: "I'm sorry, but I can't provide the explanation you're asking for because the selected code you're referring to hasn't been provided. Could you please provide the code snippet you want explained?"
-File reference is still in chatbox in chat panel
-enter "/explain" and send in chat panel
-Chat Panel result: "The provided code appears to be written in ABAP (Advanced Business Application Programming), a high-level programming language created by the German software company SAP..."

1b Screenshots:
no screenshots, sorry

1c Copilot4Eclipse Status Dialog:
Copilot Session Status
User: <redacted>
Connection: OK
Status: Copilot4Eclipse ready
Copilot API Version: 1.180.0

1d Eclipse error log:
for the case of the right-click-context-menu action not working:
[Copilot4Eclipse_no_context errorlog redacted.log](https://github.com/user-attachments/files/16798598/Copilot4Eclipse_no_context.errorlog.redacted.log)

Chatlog can't be included because it contains internal data

for the context being available after fiddling with the file reference in the chat panel:
[Copilot4Eclipse working after fiddling errorlog redacted.log](https://github.com/user-attachments/files/16798603/Copilot4Eclipse.working.after.fiddling.errorlog.redacted.log)

Chatlog can't be included because it contains internal data

2. Details about your Eclipse environment:
![eclipse version](https://github.com/user-attachments/assets/d2b64ac8-a6a9-4bb7-8b4a-9564ca945cee)
![eclipse installation details](https://github.com/user-attachments/assets/847f5a77-53da-4fe1-90b3-890be22099a0)

2a Operating System:
Edition	Windows 10 Enterprise
Version	22H2
Installed on	‎12.‎01.‎2024
OS build	19045.4780
Experience	Windows Feature Experience Pack 1000.19060.1000.0

2b Eclipse version:
Version: 2024-06 (4.32.0)
Build id: 20240606-1231

2c Java distribution and version used by eclipse:
java.version=21.0.4
java.vendor=Eclipse Adoptium

3 Are you installing or operating Eclipse from behind a corporate network proxy or firewall:
eclipse is operated in a corporate network, but no issues related to that have occured so far, and the issue in this report does not seem to be related to proxy or firewall issues.

4 additional observations:
file extensions for ABAP from the Abap development Tools when using "Reference File in Chat Panel" for me are: .asinc, .asfunc, .asprog and .aclass all of these have been added to the abap languageID by me in the home folder as explained in the Copilot4Eclipse documentation.
All of these extensions show up in the Copilot4Eclipse preferences for the ABAP language.
local .abap files work as expected, however eclipse is unable to open a local .asinc file
(local as in creqated manually on my pcand opened in eclipse via drag and drop instead of received through the ABAP development tools).

![File extensions](https://github.com/user-attachments/assets/c3c8b685-a645-40cb-a0ff-eb1815124c60)

**Comment by wayneparrott** [comment]
@ansgar-maehrlein thx for sharing this issue. I have an idea of what might be goofing up. We will research this and see if we can replicate it. We will follow up as soon as we know more or request additional info.

W

**Reply by TheOriginalDavo** [reply]
Hi Wayne, any update?

**Reply by wayneparrott** [reply]
Over the year we have worked to understand the implementation variations of the few specialized editors which Copilot4Eclipse does not currently adapt to properly. Most of our adjustments so far have been to enable Copilot4Eclipse fail gracefully for such editors. In '25 we hope to revisit some of these editors which do not work well with Copilot4Eclipse to learn if there are any work-arounds we use. I can't provide a timeframe for this work as we have additional new core features in development which is consuming most of our resources.

Wayne
Copilot4Eclipse Team
